docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,577 @@
+                                  ^-----------^
+```
+
+Some options have required arguments. For example in the .NET CLI, `--output` requires a folder name argument. If the argument is not provided, the command fails.
+
+Arguments can have expected types, and the app errors if the argument can't be parsed into the expected type. For example, the following command errors because "silent" isn't one of the valid values for `--verbosity`:
+
+```dotnetcli
+dotnet build --verbosity silent
+```
+
+```output
+Cannot parse argument 'silent' for option '-v' as expected type 'Microsoft.DotNet.Cli.VerbosityOptions'. Did you mean one of the following?
+Detailed
+Diagnostic
+Minimal
+Normal
+Quiet
+```
+
+Arguments also have expectations about how many values can be provided. Examples are provided in the [section on argument arity](#argument-arity).
+
+## Order of options and arguments
+
+You can provide options before arguments or arguments before options on the command line. The following commands are equivalent:
+
+```dotnetcli
+dotnet add package System.CommandLine --prerelease
+dotnet add package --prerelease System.CommandLine 
+```
+
+Options can be specified in any order. The following commands are equivalent:
+
+```dotnetcli
+dotnet add package System.CommandLine --prerelease --no-restore --source https://api.nuget.org/v3/index.json
+dotnet add package System.CommandLine --source https://api.nuget.org/v3/index.json --no-restore --prerelease 
+```
+
+When there are multiple arguments, the order does matter. The following commands are not equivalent:
+
+```console
+myapp argument1 argument2
+myapp argument2 argument1
+```
+
+## Aliases
+
+In both POSIX and Windows, it's common for some commands and options to have aliases. These are usually short forms that are easier to type. POSIX short forms typically have a single leading hyphen followed by a single character. The following commands are equivalent:
+
+```dotnetcli
+dotnet build --verbosity quiet
+dotnet build -v quiet
+```
+
+The [GNU standard](https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html) recommends automatic aliases. That is, you can enter any part of a long-form command or option name and it will be accepted. This behavior would make the following command lines equivalent:
+
+```dotnetcli
+dotnet publish --output ./publish
+dotnet publish --outpu ./publish
+dotnet publish --outp ./publish
+dotnet publish --out ./publish
+dotnet publish --ou ./publish
+dotnet publish --o ./publish
+```
+
+`System.CommandLine` doesn't support automatic aliases.
+
+## Case sensitivity
+
+Command and option names and aliases are case-sensitive by default according to POSIX convention, and `System.CommandLine` follows this convention. If you want your CLI to be case insensitive, define aliases for the various casing alternatives. For example, `--additional-probing-path` could have aliases `--Additional-Probing-Path` and `--ADDITIONAL-PROBING-PATH`.
```

Comment:
@jonsequitur  I hate this solution and think we should have an option to be case insensitive (at some future point after GA).